data/pigui/modules/ship-internals-window.lua

```diff
@@ -131,25 +96,34 @@ local function displayShipFunctionWindow()
 	if ui.optionsWindow.isOpen then return end
 	player = Game.player
 	local current_view = Game.CurrentView()
-	local buttons = 6
-	ui.setNextWindowPos(Vector2(ui.screenWidth/2 - ui.reticuleCircleRadius - (mainButtonSize.x + 2 * mainButtonFramePadding) * buttons, ui.screenHeight - mainButtonSize.y * 3 - 8) , "Always")
-	ui.window("ShipFunctions", {"NoTitleBar", "NoResize", "NoFocusOnAppearing", "NoBringToFrontOnFocus", "NoSavedSettings"},
+	local buttons = 3
+	local thrust_widget_size = Vector2(mainButtonSize.x * 3, mainButtonSize.y * 2)
+	assert(thrust_widget_size.y >= mainButtonSize.y)
+	local window_width = ui.getWindowPadding().x * 2 + (mainButtonSize.x + 2 * mainButtonFramePadding + ui.getItemSpacing().x) * buttons + thrust_widget_size.x + mainButtonFramePadding * 2
+	local window_height = thrust_widget_size.y + mainButtonFramePadding * 2 + ui.getWindowPadding().y * 2
+	local window_posx = ui.screenWidth/2 - ui.reticuleCircleRadius - window_width + 12 -- manual move a little closer to the center
+	local window_posy = ui.screenHeight - window_height
+	ui.setNextWindowPos(Vector2(window_posx, window_posy), "Always")
+	ui.window("ShipFunctions", {"NoTitleBar", "NoResize", "NoFocusOnAppearing", "NoBringToFrontOnFocus", "NoSavedSettings", "AlwaysAutoResize"},
 						function()
 							if current_view == "world" then
-								local v = ui.getCursorPos()
-								ui.setCursorPos(Vector2(0, v.y + 1.33 * mainButtonSize.y));
+								local shift = Vector2(0.0, thrust_widget_size.y - mainButtonSize.y)
+								ui.setCursorPos(ui.getCursorPos() + shift)
 								button_wheelstate()
+								ui.sameLine()
 								button_rotation_damping()
 								ui.sameLine()
 								button_lowThrustPower()
-								button_thrustIndicator()
+								ui.sameLine()
+								ui.setCursorPos(ui.getCursorPos() - shift)
+								button_thrustIndicator(thrust_widget_size)
 								if ui.noModifierHeld() and ui.isKeyReleased(ui.keys.f8) then
                                     show_thrust_slider = not show_thrust_slider
 								end
 							end -- current_view == "world"
-	end)
+						end)
```

Comment:
I'd prefer that the entire function is de-indented to a single tab per block change. I know this whitespace change is probably due to the editor, but incremental cleanups are a great help overall.

Reply:
When the PR is in final form, I will add a commit with indentation of all such files.